components/new-contribution-flow/index.js

```diff
@@ -284,20 +285,31 @@ class ContributionFlow extends React.Component {
     }
 
     // Check that the contributor is not blocked from contributing to the collective
-    const collectiveRejectedCategories = get(this.props.collective, 'settings.moderation.rejectedCategories', []);
-    if (collectiveRejectedCategories) {
-      const contributorRejectedCategories = get(stepProfile, 'data.categories', []);
-      const containsRejectedCategories = intersection(collectiveRejectedCategories, contributorRejectedCategories);
-      if (!isEmpty(containsRejectedCategories)) {
-        this.setState({
-          stepProfile: { ...this.state.stepProfile, contributorRejectedCategories: containsRejectedCategories },
-        });
-      }
+    const containsRejectedCategories = this.getContributorRejectedCategories(stepProfile);
+    if (!isEmpty(containsRejectedCategories)) {
+      this.setState({
+        stepProfile: { ...this.state.stepProfile, contributorRejectedCategories: containsRejectedCategories },
+      });
     }
 
     return true;
   };
 
+  getContributorRejectedCategories = account => {
+    const rejectedCategories = get(this.props.collective, 'settings.moderation.rejectedCategories', []);
```

Comment:
We also don't appear to be using this variable at any point? Or filtering the contributor's blocked categories against the collective's? We only compare the contributor's blocked categories against the list of moderation categories which would just confirm whether it's on our list or not.

Reply:
We use the method line 288 https://github.com/opencollective/opencollective-frontend/pull/5183/files#diff-fa5631eef3cde829eac1496d2cdeacc7R288

Reply:
No I mean we don't use the variable `rejectedCategories` on line 299 at any point. I can see that the `getContributorRejectedCategories` is used on line 288.

I'm asking because when I log out what is returned from the filtered `contributorRejectedCategories` on line 306, it is an empty array, which is not correct, because I have some categories blocked on my local test environment that should be showing up.

So I'm asking because it doesn't seem to be working and we also don't appear to be _using_ the collective's rejected categories at any point (`rejectedCategories` on line 299).

Reply:
Yes, something was definitely wrong.

Reply:
I updated yesterday night and added some comments. Sorry again about this complication. Happy to talk about it.